qapi/protocols/cryptography/key_distribution/cpp/IntArray.cpp

```diff
@@ -25,7 +25,14 @@ class IntArray {
           // Allocates new memory so must be freed
           int* random_array = new int[size];
           for (int i = 0; i < size; i++) {
-            random_array[i] = rand() % 2;
+            try
+            {
+              std::random_device rd;
```

Comment:
Move std::random device outside of the loop, calling it multiple times is redundant.

Reply:
See [here](https://en.cppreference.com/w/cpp/numeric/random/random_device) for an example of it being used. Otp we discussed and I read up some more on it.
std::random_device can fail
- On construction, ie `std::random_device [name]`, throwing an exception and exiting the whole program.
- Or when requesting a random number, ie `rd() % 2`. Currently your code would catch this and use srand() in each spot.

I think you have 2 options here. One is the easy way out and the other is more proper/more involved, but whichever you want to do is fine with me:
- For our purposes it's fine if the whole program exits with an exception if `std::random_device` fails.
-  Possibly, a better way to construct this is to have 2 separate classes, one `IntArraySrand` and `IntArrayRandDevice` or something like that, and in a parent class `IntArray` attempt to build the array using `IntArrayRandDevice` and if that fails, build it using `IntArraySrand`. This way you could have `std::random_device` contained within the `IntArrayRandDevice` class and so if it fails when constructing the class you know you can't use it anymore.

Reply:
The second option using the classes would allow for easier reuse later; notably, being able to combine the generators in different ways for different purposes that appeal better to the different rngs' strengths.